Slim/Player/Protocols/HTTP.pm

```diff
@@ -698,8 +698,11 @@ sub requestString {
 	# Although the port can be part of the Host: header, some hosts (such
 	# as online.wsj.com don't like it, and will infinitely redirect.
 	# According to the spec, http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
-	# The port is optional if it's 80, so follow that rule.
-	my $host = $port == 80 ? $server : "$server:$port";
+	# The port is optional if it's default to the service requested, so follow that rule.
+	my $host = $server;
+	if ( $port != ( 80 || 443 ) ) {
```

Comment:
I don't think this test is correct. You probably mean
```
$host .= ":$port" if $port !~ /(80|443)/;
or
$host .= ":$port" if $port != 80 && $port != 443;
```

Reply:
I was trying to mirror [this bit of code](https://github.com/Logitech/slimserver/blob/public/8.1/Slim/Networking/Async/HTTP.pm#L250) as closely as possible. I do not know perl very well, but this worked in a simple test of mine.

Which variant do you think is better? I can update the PR accordingly.

Reply:
Well, I would then simply keep the original line as I think the intention is simply to not add port when using HTTPS, right? (I've not followed the whole discussion on the forum)
```
my $host = ($port == 80 || $port == 443) ? $server : "$server:$port";
```

Reply:
@philippe44's version looks better. I'll do that.

BTW: the comment in the referenced code was even more interesting... https://github.com/Logitech/slimserver/blob/fde5c7622c426708524f5f882c784031c5d02a60/Slim/Networking/Async/HTTP.pm#L247